src/test/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseActionTest.java

```diff
@@ -0,0 +1,279 @@
+  Path filePath;
+  DialogHelper<ModuleSelectionViewModel> moduleSelectionDialogHelper;
+  DialogHelper<SubmissionViewModel> submissionDialogHelper;
+  Dialogs.Factory<ModuleSelectionViewModel> moduleSelectionDialogFactory;
+  Dialogs.Factory<SubmissionViewModel> submissionDialogFactory;
+  MainViewModelProvider mainVmProvider;
+  FileFinder fileFinder;
+  SubmitExerciseAction.ModuleSource moduleSource;
+  Dialogs dialogs;
+  Notifier notifier;
+  AnActionEvent event;
+  SubmitExerciseAction action;
+
+  /**
+   * Called before each test.
+   *
+   * @throws IOException Never.
+   * @throws FileDoesNotExistException Never.
+   */
+  @SuppressWarnings("unchecked")
+  @Before
+  public void setUp() throws IOException, FileDoesNotExistException {
+    course = new ModelExtensions.TestCourse("91");
+    exercise = new Exercise(12, "Test exercise");
+    group = new Group(124, Collections.singletonList("Only you"));
+    groups = Collections.singletonList(group);
+    exerciseGroup = new ExerciseGroup("Test EG", Collections.singletonList(exercise));
+    exerciseGroups = Collections.singletonList(exerciseGroup);
+    fileName = "some_file.scala";
+    fileKey = "file1";
+    file = new SubmittableFile(fileKey, fileName);
+    submissionInfo = new SubmissionInfo(1, new SubmittableFile[]{file});
+    exerciseDataSource = spy(new ModelExtensions.TestExerciseDataSource());
+    doReturn(groups).when(exerciseDataSource).getGroups(course);
+    doReturn(submissionInfo).when(exerciseDataSource).getSubmissionInfo(exercise);
+    doReturn(exerciseGroups).when(exerciseDataSource).getExerciseGroups(course);
+    main = new Main(exerciseDataSource);
+
+    mainViewModel = new MainViewModel(main);
+    courseViewModel = new CourseViewModel(course);
+    mainViewModel.courseViewModel.set(courseViewModel);
+    exercises = new ExercisesTreeViewModel(Collections.singletonList(exerciseGroup));
+    mainViewModel.exercisesViewModel.set(exercises);
+
+    exercises.getGroupViewModels().get(0).getExerciseViewModels().get(0).setSelected(true);
+
+    moduleName = "MyModule";
+    modulePath = Paths.get(moduleName);
+    moduleFilePath = modulePath.resolve("MyModule.iml").toString();
+    module = mock(Module.class);
+    doReturn(moduleFilePath).when(module).getModuleFilePath();
+    doReturn(moduleName).when(module).getName();
+
+    project = mock(Project.class);
+
+    filePath = modulePath.resolve(fileName);
+
+    mainVmProvider = mock(MainViewModelProvider.class);
+    doReturn(mainViewModel).when(mainVmProvider).getMainViewModel(project);
+
+    fileFinder = mock(FileFinder.class);
+    doReturn(filePath).when(fileFinder).tryFindFile(modulePath, fileName);
+    doCallRealMethod().when(fileFinder).findFile(any(), any());
+    doCallRealMethod().when(fileFinder).findFiles(any(), any());
+
+    moduleSource = mock(SubmitExerciseAction.ModuleSource.class);
+    doReturn(new Module[] { module }).when(moduleSource).getModules(project);
+
+    dialogs = new Dialogs();
```

Comment:
It looks like the assertions about which dialogs are shown have been removed, is there a reason for that?